### PR TITLE
Sphinx document for API references

### PR DESCRIPTION
**Pull request by chezou**
This is a follow up for #65 

- Add autodoc based contents for rest of the modules
- Ensure Google style

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -10,9 +10,69 @@
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 #
-# import os
-# import sys
-# sys.path.insert(0, os.path.abspath('.'))
+import os
+import sys
+sys.path.insert(0, os.path.abspath('../tdclient'))
```
Comment (chezou) [inline docs/conf.py]: @aodag Not sure if it is a mandatory or not for autodoc. Could you give me an advice?

File: docs/gettingstarted.rst [hunk 2]
```diff
@@ -0,0 +1,141 @@
+Requirements
```
Comment (chezou) [inline docs/gettingstarted.rst]: `gettingstarted.rst` is introduced by converting README.md to .rst manually. Not sure this is the best way to manage two documents, though.

File: docs/conf.py (hunk 1 above)
Comment (aodag) [reply docs/conf.py]: if we can't install tdclient on building docs, this workaround is needed including to PYTHONPATH.

Comment (chezou) [reply docs/conf.py]: Thanks, removed e59d7a5

File: docs/gettingstarted.rst (hunk 2 above)
Comment (takuti) [reply docs/gettingstarted.rst]: This way, README and documentation site may become inconsistent and possibly cause confusion. Don't you directly replace `README.md` with `README.rst` and embed it to the `gettingstarted` page as follows? 

```
.. include:: ../README.rst
```

Comment (chezou) [reply docs/gettingstarted.rst]: I know that option, or we can use m2r to include markdown.
anyway, this is a Python project, I'll change README into rst format.

Comment (chezou) [reply docs/gettingstarted.rst]: Converted to rst format f49d516

Comment (aodag) [reply docs/gettingstarted.rst]: @chezou long_description in setup.py should be change to README.rst
https://github.com/treasure-data/td-client-python/blob/f49d5163a0654875310602c676ef2024b3d32a88/setup.py#L36

Comment (chezou) [reply docs/gettingstarted.rst]: @aodag Good catch! 4b7d687

File: docs/index.rst [hunk 3]
```diff
@@ -1,16 +1,21 @@
-.. td-client-python documentation master file, created by
-   sphinx-quickstart on Fri Sep  6 10:12:53 2019.
-   You can adapt this file completely to your liking, but it should at least
-   contain the root `toctree` directive.
+td-client - Treasure Data API library for Python
+========================================================
+
+``td-client`` is a Python client for Arm Treasure Data's REST API.
```
Comment (takuti) [inline docs/index.rst]: Do we really need this (and following) line? The generated index page repeatedly explains the same statement...
Screen Shot 2019-09-27 at 9 41 56 AM

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+   import tdclient
+
+   def on_waiting(cursor):
+       print(cursor.job_status())
+
+   with tdclient.connect(db="sample_datasets", type="presto", wait_callback=on_waiting) as td:
+       data = pandas.read_sql("SELECT symbol, COUNT(1) AS c FROM nasdaq GROUP BY symbol", td)
+       print(repr(data))
+
+We offer another package for pandas named `pandas-td <https://github.com/treasure-data/pandas-td>`_ with some advanced features.
```
Comment (takuti) [inline README.rst]: Let's replace pandas-td with pytd.

File: docs/index.rst (hunk 3 above)
Comment (chezou) [reply docs/index.rst]: fixed f994fd5

File: README.rst (hunk 4 above)
Comment (chezou) [reply README.rst]: indeed 474b77a

**Comment by chezou** [comment]
@takuti In my local environment, I couldn't find any warning. I think all syntax issues on docstring were fixed on https://github.com/treasure-data/td-client-python/pull/70/commits/d1cedaf7100d89604062c08a942812484f9c58cc but could you give me specific examples of warning?

**Comment by takuti** [comment]
Sorry, my bad - I forgot to update local `td-client` version before generating the Sphinx doc, and the doc site has been generated based on the docstrings in the PyPI version...

Now it works fine:

```
pip uninstall td-client -y && pip install .
cd docs
make clean && make html
```

**Pull request merged**